CONCURRENCY.md

```diff
@@ -0,0 +1,102 @@
+### Concurrency in Kotlin/Native
+
+  Kotlin/Native runitime doesn't encourage classical thread-oriented concurrency
+ model with mutually exclusive code blocks and conditional variables, as this model is
+ known to be error-prone and unreliable. Instead, we suggest collection of
+ alternative approaches, allowing to use hardware concurrency and implement blocking IO.
+ Those approaches are as following, and will be elaborated in further sections:
+   * Workers with message passing
+   * Object subgraph ownership transfer
+   * Object subgraph freezing
+   * Object subgraph detachment
+   * Raw shared memory using C globals
+
+ ## Workers
+
+  Instead of threads Kotlin/Native runtime offers concept of workers: concurrently executing
+ control flow streams with an associated request queue. Workers are very similar to actors
+ in Actor Model. Worker can exchange Kotlin objects with other workers, so that at the moment
+ each mutable object is owned by the single worker, but ownership could be transferred.
+ See section [Object transfer and freezing](#transfer).
+
+  Once worker is started with `startWorker` function call, it can be uniquely addressed with an integer
+ worker id. Other workers, or non-worker concurrency primitives, such as OS threads, could send a message
+ to the worker with `schedule` call.
+ ```kotlin
+   val future = schedule(TransferMode.CHECKED, { SomeDataForWorker() }) {
+       // data returned by the second function argument comes to the
+       // worker routine as 'input' parameter.
+       input ->
```

Comment:
Following Kotlin code formatting conventions would make this code more readable.

Reply:
IDEA believes this code is properly formatted.

Reply:
Yes, you are right. Moreover, this formatting seems to be explicitly recommended for some cases: https://kotlinlang.org/docs/reference/coding-conventions.html#lambda-formatting